django/db/models/base.py

```diff
@@ -1108,6 +1108,26 @@ def _prepare_related_fields_for_save(self, operation_name, fields=None):
                 ):
                     field.delete_cached_value(self)
 
+        # GenericForeignKey fields are private
+        for field in self._meta.private_fields:
+            if (
+                field.is_relation
+                and field.is_cached(self)
+                and hasattr(field, "fk_field")
+            ):
+                obj = field.get_cached_value(self, default=None)
+                if not obj:
+                    continue
+                if obj.pk is None:
+                    raise ValueError(
+                        "%s() prohibited to prevent data loss due to unsaved "
+                        "related object '%s'." % (operation_name, field.name)
```

Comment:
We can use a f-string
```suggestion
                        f"{operation_name}() prohibited to prevent data loss due to "
                        f"unsaved related object '{field.name}'."
```